### DISCUSSION TITLE
[firebase_core] Allow initialization of default Firebase apps from `FirebaseOptions`

### DISCUSSION THREAD

**Discussion by Pante** [firebase/flutterfire#7232] [Feature request]
At the moment, initialization of Firebase application instances is (in my opinion) less than ideal. 

To initialize the main application, a `google-services.json` & `GoogleService.plist` needs to be placed in the respective folders. It cannot be initialized using a `FirebaseOptions`. On the other hand, initialization of secondary apps require a `FirebaseOptions`, and there is no in-built mechanism to retrieve the values from a configuration file. 

This creates a dichotomy between the initialization of default and secondary apps, which is frankly, frustrating to work with. Developers are forced to manage the creation of default and secondary apps differently.

I believe the following will alleviate the issue greatly:
* Allow the default application to be created using `FirebaseOptions`
* Provide a mechanism to load secondary applications from configuration files.

**Comment by lupiter-dev** [comment]
I've also added a similar request quire some time ago: [3926](https://github.com/FirebaseExtended/flutterfire/discussions/3926). Unfortunately, no progress there...Probably other topics are more important.

**Reply by Pante** [reply]
> 
> 
> I've also added a similar request quire some time ago: [3926](https://github.com/FirebaseExtended/flutterfire/discussions/3926). Unfortunately, no progress there...Probably other topics are more important.

I just read through it but I don't understand the "it will make it complicated for new users" rationale. Allowing the option to specify a `FirebaseOptions` for the default instance while allowing it to still be initialized with the configuration file by default doesn't seem to change anything for new users.

**Reply by lupiter-dev** [reply]
I think, he wanted to explain, why the current default workflow is using the configuration file. It's easier for new users, because the file is generated by Firebase and you don't have to extract values manually.
